### ISSUE TITLE
how to display evaluation metrics such as confusion matrix, mAP@50, and others for the test dataset using a YOLO model.

### ISSUE THREAD

**Issue by DhimasDwiNugroho** [ultralytics/ultralytics#18777]
### Search before asking

- [x] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/orgs/ultralytics/discussions) and found no similar questions.

### Question

I would like to ask about how to display evaluation metrics such as confusion matrix, mAP@50, and others for the test dataset using a YOLO model. My dataset is divided into three parts: training, validation, and testing. The evaluation results for training and validation are already available, but I am unsure how to display the evaluation results for the testing dataset. Could you please guide me on how to achieve this? Additionally, I would appreciate it if you could provide the detailed steps to perform this process. Thank you."

**Comment by UltralyticsAssistant** [comment]
👋 Hello @DhimasDwiNugroho, thank you for your interest in Ultralytics 🚀! We recommend visiting the [Docs](https://docs.ultralytics.com/) where you can find many helpful resources for [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) usage, which might already address your question.

If this is a 🐛 Bug Report, please provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum-reproducible-example/) to assist us in debugging it.

If this is a ❓ Question related to custom datasets, metrics, or testing, please ensure you've provided as much detail as possible, such as your dataset structure, testing logs, or any relevant customizations you've made. Additionally, check out our [Guide to Evaluation Metrics](https://docs.ultralytics.com/tasks/detection/#metrics) for further clarification on metrics like confusion matrix and mAP.

To evaluate a YOLO model on a test dataset, you can try running the following command (if not already):

```bash
yolo val data=<dataset_config>.yaml split=test
```

Ensure your test set configuration is correctly set up in your dataset `.yaml` file under the `test:` path.

Join the Ultralytics community for more discussions and real-time support:
- [Discord](https://discord.com/invite/ultralytics) 🎧
- [Discourse](https://community.ultralytics.com/)
- [Subreddit](https://reddit.com/r/Ultralytics)

## Upgrade

Ensure you're using the newest version of `ultralytics` to rule out any potential version-related issues. You can update the package with:

```bash
pip install -U ultralytics
```

## Environments

YOLO operates smoothly in the following verified environments (ensure all dependencies, including [CUDA](https://developer.nvidia.com/cuda-zone)/[CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/), and [PyTorch](https://pytorch.org/), are up to date):

- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule) tests are currently passing. CI tests verify correct operation of all YOLO [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) on macOS, Windows, and Ubuntu every 24 hours and on every commit.

This is an automated response, but rest assured that an Ultralytics engineer will assist you shortly 😊

**Comment by Y-T-G** [comment]
You can run `model.val(split="test")`
